terraform/modules/google/compute/nodejs/nodejs.sh.tpl

```diff
@@ -0,0 +1,205 @@
+#!/bin/bash
+
+NAME="${node_name}-$(hostname)"
+SANITIZEDNAME=$${NAME//-/_} # Replace hyphens with underscores, Consul Template doesn't like hyphens
+SSLCERTDIR=/usr/local/etc
+SSLSITECERTPATH=$SSLCERTDIR/site.crt
+SSLVAULTCERTPATH=$SSLCERTDIR/vault.crt
+NODEJSPOLICYNAME=${vault_policy}
+NODEJSPOLICY=/opt/vault/policies/$NODEJSPOLICYNAME.json
+GENERICSECRETPATH=secret/$NODEJSPOLICYNAME/$SANITIZEDNAME
+GENERICSECRETKEY=secret_key
+GENERICSECRET="This is a secret stored in Vault for $NAME using the $NODEJSPOLICYNAME policy"
+VAULT=https://vault.service.consul:8200
+CONSUL=http://127.0.0.1:8500
+LOGS=/var/log/user_data.log
+
+logger() {
+  DT=$(date '+%Y/%m/%d %H:%M:%S')
+  echo "$DT nodejs.sh: $1" | sudo tee -a $LOGS > /dev/null
+}
+
+preparepolicy() {
+  FILEPATH=$1
+  PARAMETER=$2
+
+  sed -i -- 's/\"/\\"/g' $FILEPATH
+  sed -i '1s/^/{"{{ parameter }}": "/' $FILEPATH
+  sed -i '$s|$|"}|' $FILEPATH
+  sed -i -- "s/{{ parameter }}/$PARAMETER/g" $FILEPATH
+}
+
+logger "Configuring Consul..."
+
+sed -i -- "s/{{ atlas_username }}/${atlas_username}/g" /etc/consul.d/base.json
+sed -i -- "s/{{ atlas_environment }}/${atlas_environment}/g" /etc/consul.d/base.json
+sed -i -- "s/{{ atlas_token }}/${atlas_token}/g" /etc/consul.d/base.json
+sed -i -- "s/{{ datacenter }}/${atlas_environment}/g" /etc/consul.d/base.json
+sed -i -- "s/{{ node_name }}/${node_name}/g" /etc/consul.d/base.json
+sed -i -- "s/{{ deploy }}/${deploy}/g" /etc/consul.d/nodejs.json
+
+service consul restart
+
+logger "Updating certs..."
```

Comment:
Is the Vault stuff included? If not, we can probably ditch everything after this line aside from properly starting nodejs.